src/ocean/core/Exception.d

```diff
@@ -198,53 +197,40 @@ public template ReusableExceptionImplementation()
 
         Intended to be used in hosts that do dynamic formatting of the
         error message and want to avoid repeating allocation with help
-        of ReusableException semantics
+        of ReusableException semantics. Support appending formatted message with
+        the help of `ocean.text.convert.Formatter`.
 
-        Leaves other fields untouched
+        Leaves other fields untouched.
 
         Params:
-            msg = string to append to the message
+            arg  = format of string to append to the message OR single integer
+                value to append
+            args = variadic arguments to be used when formatting, valid only if
+                first argument is string
 
         Returns:
             this instance
 
     ***************************************************************************/
 
-    public typeof (this) append ( cstring msg )
+    public typeof (this) append (T, Args...) ( T arg, Args args )
     {
-        ocean.core.array.Mutation.append(this.reused_msg, msg);
-        return this;
-    }
-
-    /**************************************************************************
-
-        Appends an integer to mutable exception message
-
-        Does not cause any appenditional allocations
-
-        Params:
-            num = number to be formatted
-            hex = optional, indicates that value needs to be formatted as hex
-
-        Returns:
-            this instance
-
-    **************************************************************************/
-
-    public typeof (this) append ( long num, bool hex = false )
-    {
-        char[long.max.stringof.length + 1] buff;
-        if (hex)
+        static if (is(T : long) && args.length <= 1)
+        {
+            istring fmt;
+            static if (args.length == 1 && is(Args[0] == bool))
+                fmt = args[0] ? "0x{:X}" : "{}";
+            else
+                fmt = "{}";
+            ocean.text.convert.Formatter.sformat(this.reused_msg, fmt, arg);
+        }
+        else static if (is(T : cstring))
```

Comment:
Not so fan of mixing semantics though...
`append!(cstring, cstring)` => format
`append!(uint, cstring)` => append

That is a weird API to me.

Reply:
It was not intended as a feature - can't overload two different templated `append` methods in D1 but must preserve backwards compatibility. It is one of stupid legacy quirks that will have to wait for full D2 :(

Reply:
I would much prefer to introduce a new symbol then... Maybe `format` ?

Reply:
Sure, renamed to `fmtAppend` and restored old methods as-is. Ugly but I see your point.